tests/e2e/framework/rawVM.go

```diff
@@ -0,0 +1,243 @@
+	o, e := u.Shell(cmd)
+	return strings.Trim(o, "\n"), e
+}
+
+// GetExternalIP returns the internal IP of the VM
+func (vm *GCPRawVM) GetExternalIP() (string, error) {
+	cmd := vm.baseCommand("describe") +
+		" --format='value(networkInterfaces[0].networkIP)'"
+	o, e := u.Shell(cmd)
+	return strings.Trim(o, "\n"), e
+}
+
+// SecureShell execeutes cmd on vm through ssh
+func (vm *GCPRawVM) SecureShell(cmd string) (string, error) {
+	ssh := fmt.Sprintf("gcloud compute ssh --project %s --zone %s %s --command \"%s\"",
+		vm.ProjectID, vm.Zone, vm.Name, cmd)
+	return u.Shell(ssh)
+}
+
+// SecureCopy copies files to vm via scp
+func (vm *GCPRawVM) SecureCopy(files ...string) (string, error) {
+	filesStr := strings.Join(files, " ")
+	scp := fmt.Sprintf("gcloud compute scp --project %s --zone %s %s %s",
+		vm.ProjectID, vm.Zone, filesStr, vm.Name)
+	return u.Shell(scp)
+}
+
+// Teardown releases the VM to resource manager
+func (vm *GCPRawVM) Teardown() error {
+	for _, yaml := range kubeDeleteYaml {
+		cmd := fmt.Sprintf("cat <<EOF | kubectl delete -f -\n%sEOF", yaml)
+		if _, err := u.Shell(cmd); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Setup initialize the VM
+func (vm *GCPRawVM) Setup() error {
+	if err := prepareConstants(); err != nil {
+		return err
+	}
+	if err := vm.provision(); err != nil {
+		return err
+	}
+	if err := vm.prepareCluster(); err != nil {
+		return err
+	}
+	if err := vm.setupMeshEx("generateDnsmasq"); err != nil {
+		return err
+	}
+	if err := vm.setupMeshEx("generateClusterEnv", vm.ClusterName); err != nil {
+		return err
+	}
+	return vm.setupMeshEx("machineSetup", vm.Name)
+	// TODO (chx) uses release deb pkg to install istio for now. Need to build and use latest image for every PR.
+}
+
+func (vm *GCPRawVM) provision() error {
+	// check if VM has been created
+	if _, err := u.ShellMuteOutput(vm.baseCommand("describe")); err != nil {
+		createVMcmd := vm.baseCommand("create") + fmt.Sprintf(
+			`--machine-type "n1-standard-1" \
+			 --subnet default \
+			 --can-ip-forward \
+			 --service-account %s \
+			 --scopes "https://www.googleapis.com/auth/cloud-platform" \
+			 --tags "http-server","https-server" \
+			 --image %s \
+			 --image-project %s \
+			 --boot-disk-size "10" \
+			 --boot-disk-type "pd-standard" \
+			 --boot-disk-device-name "debtest"`,
+			vm.ServiceAccount, vm.Image, vm.ImageProject)
+		if _, err = u.Shell(createVMcmd); err != nil {
+			return err
+		}
+	} else if _, err := u.Shell(vm.baseCommand("reset")); err != nil {
+		// reset existing VM
+		// TODO (chx) the GCP reset might not give us a clean slate as we assumed.
+		// 			Pending investigation and test env preprovisioning, might be better off
+		//			deleting and creating the VM.
+		return err
+	}
+	if _, err := u.Shell("gcloud compute firewall-rules describe allow-vm-ssh-http"); err != nil {
+		// create irewall rule that allow access to vm
+		if _, err = u.Shell(`gcloud compute firewall-rules create allow-vm-ssh-http \
```

Comment:
I don't think that's really needed

Reply:
It appears that this firewall rule gets deleted after some time and might just be another GCP issue. Having this logic ensures we have access to the vm.